### ISSUE TITLE
Problems compiling

### ISSUE THREAD

**Issue by henfri** [sahib/rmlint#404]
Hello,

I cannot compile rmlint on debian 9:
```
Compiling ==> lib/cfg.c
lib/cfg.c: In function 'rm_cfg_add_path':
lib/cfg.c:130:21: warning: implicit declaration of function 'g_canonicalize_filename' [-Wimplicit-function-declaration]
         real_path = g_canonicalize_filename(path, NULL);
                     ^~~~~~~~~~~~~~~~~~~~~~~
lib/cfg.c:130:19: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
         real_path = g_canonicalize_filename(path, NULL);
                   ^
lib/cfg.c: At top level:
cc1: warning: unrecognized command line option '-Wno-implicit-fallthrough'
Compiling ==> lib/xattr.c

...

==> OK!
Linking Static Library ==> librmlint.a
Ranlib Library ==> librmlint.a
Linking Program ==> rmlint
librmlint.a(cfg.o): In function `rm_cfg_add_path':
cfg.c:(.text+0x498): undefined reference to `g_canonicalize_filename'
collect2: error: ld returned 1 exit status
scons: *** [rmlint] Error 1
scons: building terminated because of errors.
root@homeserver:~/rmlint/rmlint#
```

Can you give me a hint?

Regards,
Hendrik

**Comment by sahib** [comment]
This is because of the pretty old software that Debian ships with. I've used a method (``g_canonicalize_filename()``) in the ``develop`` branch (which you are apparently using, ``master`` would have worked) that is only available in GLib 2.58 (available since Jan 2018).

Since too many people still use those old versions, I copied that util to the rmlint source tree for now - in the hope to remove it in a few years for good. So please try again.

**Comment by henfri** [comment]
That worked. Thanks!
